server/protocols/jmap-rfc-8621/src/main/scala/org/apache/james/jmap/json/EmailQuerySerializer.scala

```diff
@@ -43,6 +44,29 @@ class EmailQuerySerializer @Inject()(mailboxIdFactory: MailboxId.Factory) {
   private implicit val limitFormat: Format[Limit] = Json.valueFormat[Limit]
   private implicit val messageIdWrites: Writes[MessageId] = id => JsString(id.serialize())
 
+  private implicit val isAscendingReads: Reads[IsAscending] = Json.valueReads[IsAscending]
+  private implicit val isAscendingWrites: Writes[IsAscending] = Json.valueWrites[IsAscending]
+  private implicit val collationWrites: Writes[Collation] = Json.valueWrites[Collation]
+  private implicit val collationReads: Reads[Collation] = Json.valueReads[Collation]
+
+  private implicit val receivedAtSortReads: Reads[ReceivedAtSortProperty] = {
+    case JsString("receivedAt") => JsSuccess(ReceivedAtSortProperty())
+    case throwable: Throwable => JsError(throwable.getMessage)
+  }
+
+  private implicit val sortPropertyReads: Reads[SortProperty] = Json.reads[SortProperty]
+
+  private implicit val comparatorWrites: Writes[Comparator] = comparator => Json.obj(
+    "property" -> comparator.property.toSortClause.toString,
+    "isAscending" -> comparator.isAscending.get.value,
+    "collation" -> comparator.collation.getOrElse("").toString
+  )
+  private implicit val comparatorReads: Reads[Comparator] = (
```

Comment:
When debugging, i always got `NullpointerException` when `Reads[Comparator]`.
Can anyone help me a bit?

Reply:
comparator.isAscending**.get.**value

This is the root of all your issues.

Reply:
Use Json.reads[Comparator]